src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -649,10 +650,15 @@ emit_sum_vector (MonoCompile *cfg, MonoType *vector_type, MonoTypeEnum element_t
 	MonoClass *vector_class = mono_class_from_mono_type_internal (vector_type);
 	int vector_size = mono_class_value_size (vector_class, NULL);
 	int element_size;
-
-	// FIXME: Support Vector3
+	
 	guint32 nelems;
-	mini_get_simd_type_info (vector_class, &nelems);
+ 	mini_get_simd_type_info (vector_class, &nelems);
+
+	// Override nelems for Vector3, with actual number of elements
```

Comment:
```suggestion
	// Override nelems for Vector3, with actual number of elements, instead of treating it as a 4-element vector (three elements + zero).
```